main.tf

```diff
@@ -1,13 +1,77 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  // Use the provided config file path or default to the current dir
+  config_file_path = coalesce(var.config_file_path, path.cwd)
+  // Result ex: [gbl-audit.yaml, gbl-auto.yaml, gbl-dev.yaml, ...]
+  config_filenames = fileset(local.config_file_path, var.config_file_pattern)
+  // Result ex: [gbl-audit, gbl-auto, gbl-dev, ...]
+  config_files = { for f in local.config_filenames : trimsuffix(basename(f), ".yaml") => yamldecode(file("${local.config_file_path}/${f}")) }
+  // Result ex: { gbl-audit = { globals = { ... }, terraform = { project1 = { vars = ... }, project2 = { vars = ... } } } }
+  projects = { for f in keys(local.config_files) : f => lookup(local.config_files[f], "projects", {}) }
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+  environment_workspaces = {
+    for k, v in module.tfc_environment : k => {
+      "workspace" = v.workspace,
+      "projects" = [
+        for project_name, project_values in v.projects : project_name
+      ]
+    }
   }
+
+  project_workspaces = merge([
+    for k, v in module.tfc_environment : {
+      for env, config in v.projects : 
+        "${k}-${env}" => config
+    }
+  ]...)
+
+  custom_triggers = merge(flatten([
+    for k, v in local.projects : [
+      for project, settings in v.terraform : {
+        for trigger in (try(settings.triggers, null) != null ? settings.triggers : []) : 
+          "${k}-${project}-${trigger}" => {
+            source      = trigger
+            destination = "${k}-${project}"
+          } if try(settings.workspace_enabled, false)
+            # TODO - This causes issues if the workspace is created during this same execution...
+            #  && (try(local.project_workspaces[trigger].workspace, null) != null ? true : false)
+      }
+    ]
+  ])...)
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+# Create our top-level workspace
+module "tfc_config" {
+  source = "./modules/workspaces"
+
+  auto_apply            = var.config_auto_apply
+  file_triggers_enabled = true
+  name                  = "tfc-config"
```

Comment:
Lets parameterize this top-level project name

Reply:
This will be fixed in the next commit.